realm/realm-library/src/main/java/io/realm/RealmSchema.java

```diff
@@ -0,0 +1,159 @@
+ */
+public final class RealmSchema {
+
+    private static final String TABLE_PREFIX = Table.TABLE_PREFIX;
+    private static final String EMPTY_STRING_MSG = "Null or empty class names are not allowed";
+    private final ImplicitTransaction transaction;
+    private final BaseRealm realm;
+
+    /**
+     * Creates a wrapper to easily manipulate the current schema of a Realm.
+     */
+    RealmSchema(BaseRealm realm, ImplicitTransaction transaction) {
+        this.realm = realm;
+        this.transaction = transaction;
+    }
+
+    /**
+     * Returns the Realm schema for a given class.
+     *
+     * @param className name of the class
+     * @return schema object for that class or {@code null} if the class doesn't exists.
+     *
+     */
+    public RealmObjectSchema get(String className) {
+        checkEmpty(className, EMPTY_STRING_MSG);
+        String internalClassName = TABLE_PREFIX + className;
+        if (transaction.hasTable(internalClassName)) {
+            return new RealmObjectSchema(realm, transaction, transaction.getTable(internalClassName));
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Returns the {@link RealmObjectSchema} for all RealmObject classes that can be saved in this Realm.
+     *
+     * @return the set of all classes in this Realm or no RealmObject classes can be saved in the Realm.
+     */
+    public Set<RealmObjectSchema> getAll() {
+        int tableCount = (int) transaction.size();
+        Set<RealmObjectSchema> schemas = new LinkedHashSet<>(tableCount);
+        for (int i = 0; i < tableCount; i++) {
+            String tableName = transaction.getTableName(i);
+            if (Table.isMetaTable(tableName)) {
+                continue;
+            }
+            Table table = transaction.getTable(tableName);
+            schemas.add(new RealmObjectSchema(realm, transaction, table));
+        }
+        return schemas;
+    }
+
+    /**
+     * Adds a new class to the Realm.
+     *
+     * @param className name of the class.
+     * @return a Realm schema object for that class.
+     */
+    public RealmObjectSchema create(String className) {
+        checkEmpty(className, EMPTY_STRING_MSG);
+        String internalTableName = TABLE_PREFIX + className;
+        if (internalTableName.length() > Table.TABLE_MAX_LENGTH) {
+            throw new IllegalArgumentException("Class name is to long. Limit is 57 characters: " + className.length());
+        }
+        if (transaction.hasTable(internalTableName)) {
+            throw new IllegalArgumentException("Class already exists: " + className);
+        }
+        Table table = transaction.getTable(internalTableName);
+        return new RealmObjectSchema(realm, transaction, table);
+    }
+
+    /**
+     * Removes a class from the Realm. All data will be removed. Removing a class while other classes point
+     * to it will throw an {@link IllegalStateException}. Remove those classes or fields first.
```

Comment:
Yes, there is a test case for it. It is currently disabled because core crashes: https://github.com/realm/realm-core/pull/1267

You just need to remove the fields making the cycle manually (annoying, but possible).

I have been thinking about adding a "force" boolean which will automatically resolve those references and clean them, but I would be more comfortable with waiting and see if there is actually user requests for it.